pkg/lightning/backend/local/local.go

```diff
@@ -1878,13 +1878,20 @@ func (local *local) writeAndIngestByRanges(ctx context.Context, engineFile *File
 	var allErrLock sync.Mutex
 	var allErr error
 	var wg sync.WaitGroup
+	metErr := atomic.NewBool(false)
 
 	wg.Add(len(ranges))
 
-	for _, r := range ranges {
+	for i, r := range ranges {
 		startKey := r.start
 		endKey := r.end
 		w := local.rangeConcurrency.Apply()
+		// if meet error here, skip try more here to allow fail fast.
+		if metErr.Load() {
```

Comment:
this feels like something that can be handled by `errgroup` 🤔

Reply:
There are some difference, here we don't want to terminate the other running tasks if one task fails